packages/core/framework/src/build-tools/compiler.ts

```diff
@@ -140,6 +140,16 @@ export class Compiler {
     await rm(path, { recursive: true }).catch(() => {})
   }
 
+  /**
+   * Ensures a directory exists
+   */
+  async #ensureDir(path: string, clean?: boolean) {
+    if (clean) {
+      await this.#clean(path)
+    }
+    await mkdir(path, { recursive: true })
```

Comment:
Q: Would it make sense to reuse our FileSystem util? Like we create a singleton in the constructor and reuse it when necessary?

Reply:
One instance might not work, since all these operations don't share the same parent path.